crates/ibc/src/applications/transfer/coin.rs

```diff
@@ -38,6 +37,19 @@ pub struct Coin<D> {
     pub amount: Amount,
 }
 
+impl<D> Coin<D> {
+    pub fn new<T, DT>(amount: T, denom: DT) -> Self
+    where
+        T: Into<Amount>,
+        DT: Into<D>,
```

Comment:
A note regarding these bounds, as this method would get available to our users too (not only for the tests)

Although they've made improvements in cleaning up test arguments, and at first glance, it seems to give users more flexibility, it's important to note that there are situations where both `amount` and `denom` might come from either parsing a string or using `TryFrom`. This means not all conversions can simply turn into `Amount` and `D` so users may obtain them from conversions that might fail.

In fact, we even shouldn’t use `TryFrom` or similar restrictions in this specific situation. As particularly `Coin` is generic over denoms and We can't foresee how hosts will obtain this data. So any restrictions will assume a certain approach, which may not always work. Also, e.g. in cases like [here](https://github.com/cosmos/ibc-rs/blob/1aa853bb646d37aab3848096b4221450f3a2e760/crates/ibc/src/applications/transfer/coin.rs#L94-L97) and [here](https://github.com/cosmos/ibc-rs/blob/dc8e4161dd5e00a2031cf3feb58dad1ed7e26bdb/crates/ibc/src/applications/transfer/packet.rs#L46), it is easier to work with the own types.

Btw, if still want to keep our tests args as is, Perhaps we should move this method (as a helper) under the `tests` module and only have it for the `RawCoin` type. Wdyt?